pkg/client/events/replay_client.go

```diff
@@ -285,10 +294,8 @@ func (rClient *replayClient[T]) newMapEventsBytesToTFn(errCh chan<- error) func(
 				return *new(T), true
 			}
 
-			panic(fmt.Sprintf(
-				"unexpected error deserialising event: %v; eventBz: %s",
-				err, string(eventBz),
-			))
+			logger.Error().Err(err).Msgf("unexpected error deserialising event")
```

Comment:
How can we handle this error without a panic?

Reply:
Can you:

1. Explain what causes the panic?
2. Provide an example of how it can be reproduced?

Reply:
With the current code base nothing causes this panic but a future change could introduce one that would cause our gateway to panic and shutdown, or continually panic and shut down. It's a possibility I would like to look at removing, an error deserializing an event shouldn't shut down our gateway.